src/main/java/net/snowflake/ingest/connection/RequestBuilder.java

```diff
@@ -247,29 +261,26 @@ public RequestBuilder(
   public RequestBuilder(
       String accountName,
       String userName,
-      KeyPair keyPair,
+      Object credential,
       String schemeName,
       String hostName,
       int portNum,
       String userAgentSuffix,
       CloseableHttpClient httpClient,
       String clientName) {
     // none of these arguments should be null
-    if (accountName == null || userName == null || keyPair == null) {
+    if (accountName == null || userName == null || credential == null) {
       throw new IllegalArgumentException();
     }
 
     // Set up the telemetry service if needed
     this.telemetryService =
-        ENABLE_TELEMETRY_TO_SF
+        ENABLE_TELEMETRY_TO_SF && credential instanceof KeyPair
```

Comment:
Please add a TODO in the comment with JIRA mentioning that OAuth support will come later

Reply:
Fixed.